sound/soc/sof/sof-client-probes.c

```diff
@@ -223,9 +279,13 @@ static ssize_t sof_probes_dfs_points_read(struct file *file, char __user *to,
 	for (i = 0; i < num_desc; i++) {
 		offset = strlen(buf);
 		remaining = PAGE_SIZE - offset;
-		ret = snprintf(buf + offset, remaining,
-			       "Id: %#010x  Purpose: %u  Node id: %#x\n",
-				desc[i].buffer_id, desc[i].purpose, desc[i].stream_tag);
+		if (sof_client_get_ipc_type(cdev) == SOF_IPC_TYPE_4)
+			ret = sof_probes_ipc4_prints(cdev, buf + offset, remaining, &desc[i]);
+		else
+			ret = snprintf(buf + offset, remaining,
+				       "Id: %#010x  Purpose: %u  Node id: %#x\n",
+				       desc[i].buffer_id, desc[i].purpose, desc[i].stream_tag);
```

Comment:
Instead, please add a new callback to `struct sof_probes_ipc_ops`, format/decode_point_information() or something simpler.